test/distributed/_tools/test_fsdp2_mem_tracker.py

```diff
@@ -0,0 +1,300 @@
+# Owner(s): ["module: unknown"]
+import functools
+from typing import Union
+
+import torch
+import torch.nn as nn
+from torch.distributed._composable import checkpoint
+from torch.distributed._composable.fsdp import (
+    CPUOffloadPolicy,
+    fully_shard,
+    MixedPrecisionPolicy,
+    OffloadPolicy,
+)
+from torch.distributed._tensor import init_device_mesh
+from torch.distributed._tools.fsdp2_mem_tracker import FSDPMemTracker
+from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
+    apply_activation_checkpointing,
+    CheckpointWrapper,
+)
+
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest, MLP
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    ModelArgs,
+    Transformer,
+    TransformerBlock,
+)
+
+c10d_ops = torch.ops.c10d
+funcol = torch.ops.c10d_functional
+
+
```

Comment:
Test sizes are purposefully kept a bit large to stress test the tracker, but I have made sure all of them have peak memory below 3GB and run fast enough.